cli/permissions.rs

```diff
@@ -170,51 +178,89 @@ impl DenoPermissions {
   pub fn check_read(&self, filename: &str) -> DenoResult<()> {
     match self.allow_read.get_state() {
       PermissionAccessorState::Allow => Ok(()),
-      PermissionAccessorState::Ask => match self
-        .try_permissions_prompt(&format!("read access to \"{}\"", filename))
-      {
-        Err(e) => Err(e),
-        Ok(v) => {
-          self.allow_read.update_with_prompt_result(&v);
-          v.check()?;
-          Ok(())
+      state => {
+        let file_path = Path::new(filename);
+        for path in &self.read_whitelist {
+          if file_path.starts_with(path) {
+            return Ok(());
+          }
         }
-      },
-      PermissionAccessorState::Deny => Err(permission_denied()),
+        match state {
+          // This shouldn't be possible but I guess rust doesn't realize.
+          PermissionAccessorState::Allow => Ok(()),
+          PermissionAccessorState::Ask => match self
+            .try_permissions_prompt(&format!("read access to \"{}\"", filename))
+          {
+            Err(e) => Err(e),
+            Ok(v) => {
+              self.allow_read.update_with_prompt_result(&v);
+              v.check()?;
+              Ok(())
+            }
+          },
+          PermissionAccessorState::Deny => Err(permission_denied()),
+        }
+      }
     }
   }
 
   pub fn check_write(&self, filename: &str) -> DenoResult<()> {
     match self.allow_write.get_state() {
       PermissionAccessorState::Allow => Ok(()),
-      PermissionAccessorState::Ask => match self
-        .try_permissions_prompt(&format!("write access to \"{}\"", filename))
-      {
-        Err(e) => Err(e),
-        Ok(v) => {
-          self.allow_write.update_with_prompt_result(&v);
-          v.check()?;
-          Ok(())
+      state => {
+        let file_path = Path::new(filename);
+        for path in &self.write_whitelist {
+          if file_path.starts_with(path) {
+            return Ok(());
+          }
         }
```

Comment:
I think this could be combined with the read_whitelist loop into a standalone function `fn path_is_allowed(p: &Path, white_list: &HashSet): bool`, and ideally it would have some unit tests.